frontend/demo/component/treegrid/treegrid-column.ts

```diff
@@ -0,0 +1,81 @@
+import '../../init'; // hidden-full-source-line
+import '@vaadin/flow-frontend/gridConnector.js'; // hidden-full-source-line (Grid's connector)
+
+import { customElement, LitElement, query } from 'lit-element';
+import { html } from 'lit-html';
+import '@vaadin/vaadin-grid/vaadin-grid';
+import '@vaadin/vaadin-grid/vaadin-grid-tree-column';
+import '@vaadin/vaadin-ordered-layout/vaadin-horizontal-layout';
+import '@vaadin/vaadin-button/vaadin-button';
+import {
+  GridDataProviderCallback,
+  GridDataProviderParams,
+  GridElement
+} from '@vaadin/vaadin-grid/vaadin-grid';
+import { getPeople } from '../../domain/DataService';
+import Person from '../../../generated/com/vaadin/demo/domain/Person';
+import { applyTheme } from 'generated/theme';
+
+@customElement('treegrid-column')
+export class Example extends LitElement {
+  constructor() {
+    super();
+    // Apply custom theme (only supported if your app uses one)
+    applyTheme(this.shadowRoot);
+  }
+
+  private managers?: Person[];
+
+  // async firstUpdated() {
+  //   this.managers = (await getPeople({managerId: null}));
+  // }
+
+  async dataProvider(params: GridDataProviderParams, callback: GridDataProviderCallback) {
+    let people: Person[];
+
+    if (params.parentItem) {
+      const manager = params.parentItem as Person;
+      people = await getPeople({ managerId: manager.id });
+    } else {
+      people = await getPeople({ managerId: null });
+      this.managers = people;
+    }
+
+    const startIndex = params.page * params.pageSize;
+    const pageItems = people.slice(startIndex, startIndex + params.pageSize);
+    // Inform grid of the requested tree level's full size
+    const treeLevelSize = people.length;
+    callback(pageItems, treeLevelSize);
+  }
+
+  // tag::snippet[]
+  @query('vaadin-grid')
+  private grid!: GridElement;
+
+  render() {
+    return html`
+      <vaadin-horizontal-layout theme="spacing">
+        <h2 style="flex: 1; margin-bottom: 0; margin-top: 0;">Employee</h2>
+        <vaadin-button
+          @click=${() =>
+            this.managers && this.managers.forEach(manager => this.grid.expandItem(manager))}
+          >Expand All</vaadin-button
+        >
+        <vaadin-button
+          @click=${() =>
+            this.managers && this.managers.forEach(manager => this.grid.collapseItem(manager))}
```

Comment:
Same comment as above
```suggestion
          @click=${() => (this.expandedItems = [])}
```

Reply:
Changed.